include/ttmlir/Target/TTNN/program.fbs

```diff
@@ -61,6 +61,15 @@ table FullOp {
   out: tt.target.TensorRef;
 }
 
+table ArangeOp {
+  start: float;
+  end: float;
+  step: float;
+  dtype: DataType;
+  out: tt.target.TensorRef;
+  device: tt.target.DeviceRef;
```

Comment:
Might want to add a memory config field here of type `tt.target.MemoryConfigDesc` and 
`OptionalAttr<TTNN_MemoryConfigAttr>:$memory_config);` in the tablegen op definition. The goal here is to match how ttnn defines the ops and their parameters.

Reply:
Actually on second though scratch this for now... We can discuss how we want to align with TTNN later, because some TTNN ops just get all this info from the dps operand if it's provided (and we always provide the dps operand), so it'd be weird to have parameters here declared but never used.